clash-lib/src/Clash/Primitives/DSL.hs

```diff
@@ -720,6 +730,24 @@ instHO bbCtx fPos (resTy, bbResTy) argsWithTypes = do
 
   pure (TExpr resTy (Identifier resName Nothing))
 
+compInBlock
+  :: forall backend
+   . Backend backend
+  => Text
+  -> [(Text, HWType)]
+  -- ^ in ports
+  -> [(Text, HWType)]
+  -- ^ out ports
+  -> State (BlockState backend) ()
+compInBlock compName inPorts outPorts =
+
+  let inPorts' = mkPort In <$> inPorts
+      outPorts' = mkPort Out <$> outPorts
+  in addComponent $ VHDLComp compName (inPorts' ++ outPorts')
+
+ where
+  mkPort inOut (nm,ty) = (nm,inOut,ty)
```

Comment:
```suggestion
compInBlock compName inPorts0 outPorts0 =
  addComponent (VHDLComp compName (inPorts1 ++ outPorts1))
 where
  mkPort inOut (nm, ty) = (nm, inOut, ty)
  inPorts1 = mkPort In <$> inPorts0
  outPorts1 = mkPort Out <$> outPorts0
```

Mostly style changes. We generally prefer `()` over `$`, although I can't quite remember what the rationale is now :sweat_smile:. To me its easier to read, but maybe that's just because I'm used to it now.

Reply:
I find `$` easier to read 😁